### DISCUSSION TITLE
Populating inputs based on relationship values

### DISCUSSION THREAD

**Discussion by PaulMcF1987** [Budibase/budibase#7173] [Help]
I was wondering...

I have a list of customers saved, when they book a job, I select the customer from a dropdown list... The dropdown list is populated using the relationship option...

the table that the relationship column is linked to, along with the customer name, also has a lot of other info such as address etc. Currently, after selecting the customer, I then need to go on and input their address along with other information, manually.

I was wondering, is there a way to make budibase enter this information into the relevant cells.

For example... If I choose `customer A` from the relationship dropdown, that customers address details are automatically pulled from the relationship table and entered into the relevant fields?

I could write a JS onchange script which would achieve this

` if ( customer == "customer A") {
     fieldA = "B"
}`

But I would need to add every customer to this script and their details, I would then need to make multiple on change scripts to update the different fields.

Is there an option, using say, handlebars where Budibase can go into the table, find the customer and pull the info from that table and insert into the correct filed

I know that I would need to set up an onchange function for each field so that budibase knows which column to look for.

I suppose what I am looking for is something similar to how a Vlookup would work in Excel.

**Comment by melohagan** [comment]
Hi @PaulMcF1987 

I think this may be what you're looking for. This example uses
- One Manufacturer -> Many Products

but this could just as well be your Job -> Customer relationship.

Given a form for your Manufacturer with a relationship dropdown for the Products, nest the following within your form:
![Screenshot 2022-08-15 at 09 44 48](https://user-images.githubusercontent.com/101575380/184604716-95e5d41a-ee03-4571-9549-a1fcc0620eee.png)

The repeater block should point to the relationship as follows:
![Screenshot 2022-08-15 at 09 57 35](https://user-images.githubusercontent.com/101575380/184606608-b2afa6cc-5ec1-4906-a536-e86bb7682544.png)

Add this filter to your repeater block:
![Screenshot 2022-08-15 at 10 04 10](https://user-images.githubusercontent.com/101575380/184607575-9c3d1d88-57e7-4b24-8cbb-9547c6bdbd2f.png)
```js
return $("Form.Fields.Products")[0];
```

Next set the form schema to the *Product* table, and the type to *Update* (this will ensure the fields are populated)
![Screenshot 2022-08-15 at 09 58 33](https://user-images.githubusercontent.com/101575380/184606750-d8388496-708c-4531-9be9-989fcab1a4b5.png)

Add a *Field Group* to your form and click the **Update form fields** button. 

**RESULT**
![form](https://user-images.githubusercontent.com/101575380/184607866-e3d6415e-a5ec-4750-ad21-f75633719395.gif)

**Reply by PaulMcF1987** [reply]
@melohagan Thank you for your reply.
I just want to check... The `products` drop down that you are using, in my case would be `customers`...

The customers dropdown on my form is a relationship column, would the above still work?

**Reply by melohagan** [reply]
Yes that's right. The Products dropdown being used is a relationship picker. Here's screenshots of my tables as well:

![Screenshot 2022-08-15 at 11 56 00](https://user-images.githubusercontent.com/101575380/184623430-fa813e19-e6a3-459e-8433-923bee0fb43b.png)

![Screenshot 2022-08-15 at 11 56 41](https://user-images.githubusercontent.com/101575380/184623487-f6f55af3-7817-49f4-991e-8aca049d53ea.png)

**Reply by PaulMcF1987** [reply]
I have tried this...

I started with the `/new/row` page.
As with this page, there is no data provider by default, so I added a data provider and a repeater at the top before the form.

Now the editor is unresponsive, when I try to edit that page, there is a blue spinner on the screen that spins for about a minute, whilst it is spinning, I will not let me change page. After about a minute, I get the below

![image](https://user-images.githubusercontent.com/75368133/184639673-11ae9c53-113d-4bce-bff0-7a5dcc988945.png)

If I use the revert option, how far back will it go? I dont want to go any further back than the last published version

**Reply by melohagan** [reply]
Normally when I get that screen a refresh of the page fixes it. I'm guessing that didn't help in this case?
Are you sure you don't have any binding code that could be running an infinite loop?

**Reply by melohagan** [reply]
How much data are you working with? Maybe it was trying to repeat on far too many relationships - make sure the filter is added to the data provider before adding the repeater.

**Reply by PaulMcF1987** [reply]
I reverted back to the production version. Ill try it after I finish work and have more time to focus on it. I think ill be best exporting the app and having a test version to run on
